code/model/DNProject.php

```diff
@@ -148,6 +148,33 @@ public function HasDiskQuota() {
 	}
 
 	/**
+	 * Get the menu to be shown on projects
+	 *
+	 * @return ArrayList
+	 */
+	public function Menu() {
+		$list = new ArrayList();
+
+		$list->push(new ArrayData(array(
+			'Link' => sprintf('naut/project/%s', $this->Name),
+			'Title' => 'Deploy',
+			'IsActive' => Controller::curr()->getAction() == 'project'
+		)));
+
+		if(Controller::curr()->FlagSnapshotsEnabled()) {
```

Comment:
What if the current controller isn't DNRoot (e.g. in the case of a nested controller?). How about `singleton('DNRoot')` or just making it a static method?